src/handlers/rfqm_handlers.ts

```diff
@@ -0,0 +1,127 @@
+// tslint:disable:max-file-line-count
+import { InvalidAPIKeyError, NotFoundError, ValidationError, ValidationErrorCodes } from '@0x/api-utils';
+import { getTokenMetadataIfExists, isNativeSymbolOrAddress, TokenMetadata } from '@0x/token-metadata';
+import { BigNumber } from '@0x/utils';
+import * as express from 'express';
+import * as HttpStatus from 'http-status-codes';
+import { Counter } from 'prom-client';
+
+import { CHAIN_ID, NATIVE_WRAPPED_TOKEN_SYMBOL } from '../config';
+import { schemas } from '../schemas';
+import { FetchIndicativeQuoteParams, RfqmService } from '../services/rfqm_service';
+import { ConfigManager } from '../utils/config_manager';
+import { schemaUtils } from '../utils/schema_utils';
+
+const RFQM_INDICATIVE_QUOTE_REQUEST = new Counter({
+    name: 'rfqm_handler_indicative_quote_requested',
+    help: 'Request made to fetch rfqm indicative quote',
+    labelNames: ['apiKey'],
+});
+
+const RFQM_INDICATIVE_QUOTE_NOT_FOUND = new Counter({
+    name: 'rfqm_handler_indicative_quote_not_found',
+    help: 'Request to fetch rfqm indicative quote returned no quote',
+    labelNames: ['apiKey'],
+});
+
+const RFQM_INDICATIVE_QUOTE_ERROR = new Counter({
+    name: 'rfqm_handler_indicative_quote_error',
+    help: 'Request to fetch rfqm indicative quote resulted in error',
+    labelNames: ['apiKey'],
+});
+
+export class RfqmHandlers {
+    constructor(private readonly _rfqmService: RfqmService, private readonly _configManager: ConfigManager) {}
+
+    public async getIndicativeQuoteAsync(req: express.Request, res: express.Response): Promise<void> {
+        RFQM_INDICATIVE_QUOTE_REQUEST.labels(req.header('0x-api-key') || 'N/A').inc();
+        const params = this._parseFetchIndicativeQuoteParams(req);
+        let indicativeQuote;
+        try {
+            indicativeQuote = await this._rfqmService.fetchIndicativeQuoteAsync(params);
+        } catch (err) {
+            req.log.error(err, 'Encountered an error while fetching an rfqm indicative quote');
+            RFQM_INDICATIVE_QUOTE_ERROR.labels(req.header('0x-api-key') || 'N/A').inc();
+            throw new Error('Unexpected error encountered');
```

Comment:
Does this return JSON? I'd love to make sure that there is consistency in our output. If we return JSON when a response is successful, we should also try to respond with JSON when an error is encountered.

Can be as simple as:
```
{
    "error": errorMessage,
}
```

Reply:
Yup - just added a test for this specific case